File: app/src/main/java/com/awesome/photo/photoviewer/MainActivity.java
package com.awesome.photo.photoviewer;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;

import com.awesome.photo.photoviewer.databinding.ActivityMainBinding;
import com.awesome.photo.photoviewer.viewmodels.MainActivityViewModel;
import com.awesome.photo.photoviewer.viewmodels.factories.MainActivityViewModelFactory;
import com.awesome.photo.remote.api.providers.JSONPlaceHolderServiceBuilder;
import com.awesome.photo.remote.api.providers.JSONPlaceholderRepository;

public class MainActivity extends AppCompatActivity {

    MainActivityViewModel mainActivityViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mainActivityViewModel = ViewModelProviders.of(this, new MainActivityViewModelFactory(new JSONPlaceholderRepository(new JSONPlaceHolderServiceBuilder().build()))).get(MainActivityViewModel.class);
        ActivityMainBinding viewBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        //for some reason my binding isn't regenerating.
        findViewById(R.id.albums_cta).setOnClickListener(mainActivityViewModel.viewAlbumsClickListener(this));


    }
}

File: app/src/main/java/com/awesome/photo/photoviewer/MainActivity.kt
package com.awesome.photo.photoviewer

import androidx.appcompat.app.AppCompatActivity
import com.awesome.photo.photoviewer.viewmodels.MainActivityViewModel
import android.os.Bundle
import androidx.lifecycle.ViewModelProviders
import com.awesome.photo.photoviewer.viewmodels.factories.MainActivityViewModelFactory
import com.awesome.photo.remote.api.providers.JSONPlaceholderRepository
import com.awesome.photo.remote.api.providers.JSONPlaceHolderServiceBuilder
import androidx.databinding.DataBindingUtil
import com.awesome.photo.photoviewer.R
import com.awesome.photo.photoviewer.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    var mainActivityViewModel: MainActivityViewModel? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mainActivityViewModel = ViewModelProviders.of(
            this,
            MainActivityViewModelFactory(JSONPlaceholderRepository(JSONPlaceHolderServiceBuilder().build()))
        ).get(
            MainActivityViewModel::class.java
        )
        val viewBinding: ActivityMainBinding =
            DataBindingUtil.setContentView(this, R.layout.activity_main)
        viewBinding.albumsCta.setOnClickListener(
            mainActivityViewModel!!.viewAlbumsClickListener(
                this
            )
        )
    }
}